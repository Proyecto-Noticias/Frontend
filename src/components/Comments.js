import React from 'react';
import CommentItem from './CommentItem';

export default function Comments() {
  return (
    <section className='comments__box'>
      <h2>Comments</h2>
      <div className='comments__box--detail'>
        <CommentItem />
      </div>
    </section>
  );
}
