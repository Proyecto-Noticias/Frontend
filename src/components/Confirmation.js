///// START CODE REVIEW COMMENT

// Amazing!

///// END CODE REVIEW COMMENT

import React from "react";
import { Link } from "react-router-dom";

const Confirmation = () => {
  return (
    <div className="confirmation__container">
      <div className="confirmation__deco"></div>
      <div className="confirmation__text">
        <p>Your account has been activated</p>

        <Link className="confirmation__link" to="login">
          Go to login
        </Link>
      </div>
    </div>
  );
};

export default Confirmation;
