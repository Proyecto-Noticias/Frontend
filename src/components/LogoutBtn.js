import React from 'react'
import { useDispatch } from 'react-redux'
import { logout } from '../actions/auth'

export const LogoutBtn = () => {

  const dispatch = useDispatch()

  const handleLogout = () => {
    dispatch(logout())
  }

  return (
    <div>
      <button
      className='header__title-right-button'
      type='button'
        onClick={handleLogout}
      >
        Logout
      </button>
    </div>
  )
}
