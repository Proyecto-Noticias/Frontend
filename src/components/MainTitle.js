import React, { useRef, useEffect } from "react";
import { TweenMax, Power3 } from "gsap";
import { Link } from "react-router-dom";
import InstallPWA from "../components/installPWA";

const MainTitle = () => {
  let titleWithAnimation = useRef(null);
  useEffect(() => {
    TweenMax.to(titleWithAnimation, 1, {
      opacity: 1,
      y: -20,
      ease: Power3.easeOut,
    });
  }, []);

  return (
    <>
      <div
        ref={(el) => {
          titleWithAnimation = el;
        }}
        className="main__container--login">
        <h1 className="main__container--title">
          News with automated web extraction
        </h1>
        <p className="main__container--title-description">
          We use Artificial Intelligence to classify the news according to its
          positivism or negativity
        </p>
        <Link aria-label="Learn more" to="/blog">
          <button tabIndex="-1" className="main__container--btn" type="button">
            Learn more
          </button>
        </Link>
        <InstallPWA />
      </div>
    </>
  );
};

export default MainTitle;
