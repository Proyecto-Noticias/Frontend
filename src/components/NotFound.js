import React from 'react';
import { Link } from 'react-router-dom';
import bad from '../assets/negative.png';

export default function NotFound() {
  return (
    <div className='notfound__container'>
      <img src={bad} loading='lazy' alt='error 404'></img>
      <h1 className='notfound__text'>We don't find what you are looking for</h1>
      <Link to='/'>
        <button className='notfound__btn'>Go Home</button>
      </Link>
    </div>
  );
}
