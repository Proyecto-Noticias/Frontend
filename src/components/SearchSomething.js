import React from 'react'

export const SearchSomething = () => {
  return (
    <div>
      <p className='searchSomething'>Search a notice!</p> 
    </div>
  )
}
