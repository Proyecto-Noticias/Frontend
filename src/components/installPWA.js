import React, { useEffect, useState } from "react";
import styled from "styled-components";

const StyledButton = styled.button`
  background-color: transparent;
  border: none;
  font-size: 1rem;
  border-radius: 10px;
  text-decoration: none;
  color: white;
  cursor: pointer;
  &:hover {
    transform: scale(1.1);
  }
  @media (max-width: 767px) {
    font-size: 0.6rem;
  }
`;
//
const InstallPWA = () => {
  const [supportsPWA, setSupportsPWA] = useState(false);
  const [promptInstall, setPromptInstall] = useState(null);

  useEffect(() => {
    const handler = (e) => {
      e.preventDefault();
      console.log("we are being triggered :D");
      setSupportsPWA(true);
      setPromptInstall(e);
    };
    window.addEventListener("beforeinstallprompt", handler);

    return () => window.removeEventListener("transitionend", handler);
  }, []);

  const onClick = (evt) => {
    evt.preventDefault();
    if (!promptInstall) {
      return;
    }
    promptInstall.prompt();
  };
  if (!supportsPWA) {
    return null;
  }
  return <StyledButton onClick={onClick}>Install</StyledButton>;
};

export default InstallPWA;
