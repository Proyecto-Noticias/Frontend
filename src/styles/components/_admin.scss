.admin__container {
  width: 100%;
  color: $text-color;
  h1 {
    font-size: 1.75rem;
    font-weight: 400;
    text-align: center;
  }
}

.toogle--button {
  cursor: pointer;
  border: none;
  &:focus {
    border: 2px solid $ligth-color;
  }
}

.toggle--icon {
  width: 35px;
  height: 30px;
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
}

.deleteUser--button {
  @extend .toogle--button;
}
