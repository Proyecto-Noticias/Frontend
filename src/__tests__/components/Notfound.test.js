import React from 'react';
import { shallow } from "enzyme";
import { NotFound } from '../../components/NotFound';


describe('Test of <NotFound />', () => {  
  
  const wrapper = shallow( <NotFound /> )  
  
  test('should show the component correctly', () => {   
    expect ( wrapper ).toMatchSnapshot()
  })

  test('should contain className notfound__container', () => {
    const div = wrapper.find('div')
    const className = div.prop('className')

    expect( className.includes('notfound__container')).toBe( true )
  })

  it("redirects to home", () => {  
    expect(wrapper.find("Link").prop("to")).toEqual('/')
  })

  test('should have a title', () => {    
    const h1 = wrapper.find('h1')
    expect( h1.text()).toBe("We don't find what you are looking for")
  })
  
})
