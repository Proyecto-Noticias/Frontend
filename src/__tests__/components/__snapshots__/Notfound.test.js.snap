// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <NotFound /> should show the component correctly 1`] = `
<div
  className="notfound__container"
>
  <img
    alt="error 404"
    src="bad.png"
  />
  <h1
    className="notfound__text"
  >
    We don't find what you are looking for
  </h1>
  <Link
    to="/"
  >
    <button
      className="notfound__btn"
    >
      Go Home
    </button>
  </Link>
</div>
`;
